web/toolbar.js

```diff
@@ -0,0 +1,290 @@
+/* Copyright 2016 Mozilla Foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+'use strict';
+
+(function (root, factory) {
+  if (typeof define === 'function' && define.amd) {
+    define('pdfjs-web/toolbar', ['exports', 'pdfjs-web/ui_utils'],
+      factory);
+  } else if (typeof exports !== 'undefined') {
+    factory(exports, require('./ui_utils.js'));
+  } else {
+    factory((root.pdfjsWebToolbar = {}), root.pdfjsWebUIUtils);
+  }
+}(this, function (exports, uiUtils) {
+
+var mozL10n = uiUtils.mozL10n;
+var noContextMenuHandler = uiUtils.noContextMenuHandler;
+var animationStarted = uiUtils.animationStarted;
+var localized = uiUtils.localized;
+
+var DEFAULT_SCALE_VALUE = uiUtils.DEFAULT_SCALE_VALUE;
+var DEFAULT_SCALE = uiUtils.DEFAULT_SCALE;
+var MIN_SCALE = uiUtils.MIN_SCALE;
+var MAX_SCALE = uiUtils.MAX_SCALE;
+
+var PAGE_NUMBER_LOADING_INDICATOR = 'visiblePageIsLoading';
+var SCALE_SELECT_CONTAINER_PADDING = 8;
+var SCALE_SELECT_PADDING = 22;
+
+/**
+ * @typedef {Object} ToolbarOptions
+ * @property {HTMLDivElement} container - Container for the secondary toolbar.
+ * @property {HTMLSpanElement} numPages - Label that contains number of pages.
+ * @property {HTMLInputElement} pageNumber - Control for display and user input
+ *   of the current page number.
+ * @property {HTMLSpanElement} scaleSelectContainer - Container where scale
+ *   controls are placed. The width is adjusted on UI initialization.
+ * @property {HTMLSelectElement} scaleSelect - Scale selection control.
+ * @property {HTMLOptionElement} customScaleOption - The item to display of the
+ *   non-predefined scale.
+ * @property {HTMLButtonElement} previous - Button to go to the previous page.
+ * @property {HTMLButtonElement} next - Button to go to the next page.
+ * @property {HTMLButtonElement} zoomIn - Button to zoom in the current page.
```

Comment:
Nit: Given that this zooms *all* pages, perhaps `... - Button to zoom in the pages.` (and similar just below) might be more appropriate?